src/Controls/src/Core/Platform/GestureManager/GestureManager.iOS.cs

```diff
@@ -325,22 +311,53 @@ UISwipeGestureRecognizer CreateSwipeRecognizer(SwipeDirection direction, Action<
 			return result;
 		}
 
-		UITapGestureRecognizer CreateTapRecognizer(int numTaps, Action<UITapGestureRecognizer> action, int numFingers = 1)
+		UITapGestureRecognizer? CreateTapRecognizer(
+			WeakReference weakEventTracker,
+			WeakReference weakRecognizer)
 		{
+			if (!TryGetTapGestureRecognizer(weakRecognizer.Target as IGestureRecognizer, out TapGestureRecognizer? tapGesture))
+				return null;
+
+			if (tapGesture == null)
+				return null;
+
+			Action<UITapGestureRecognizer> action = new Action<UITapGestureRecognizer>((sender) =>
+			{
+				var eventTracker = weakEventTracker.Target as GestureManager;
+				var originPoint = sender.LocationInView(eventTracker?._handler?.PlatformView);
+				ProcessRecognizerHandlerTap(weakEventTracker, weakRecognizer, originPoint, (int)sender.NumberOfTapsRequired);
+			});
+
 			var result = new UITapGestureRecognizer(action)
 			{
-				NumberOfTouchesRequired = (uint)numFingers,
-				NumberOfTapsRequired = (uint)numTaps,
+				NumberOfTapsRequired = (uint)tapGesture.NumberOfTapsRequired,
 				ShouldRecognizeSimultaneously = ShouldRecognizeTapsTogether
 			};
 
+			// For whatever reason the secondary mask doesn't work on catalyst
+			// it only works when you have a mouse connected to an iPad
+			// so we just ignore setting the mask if the user is running catalyst
+			// right click is handled by adding a UIContextMenu interaction
+			if (OperatingSystem.IsIOSVersionAtLeast(13, 4) && !OperatingSystem.IsMacCatalyst())
```

Comment:
This is the other functional addition to the code. Most of the other changes in this file are organizational